webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx

```diff
@@ -112,8 +112,13 @@ export default class SubMenuItem extends React.PureComponent<Props, State> {
             } else if (action) { // leaf node in the tree handles action only
                 action(postId);
             }
-        } else if (event.currentTarget.id === id && action) {
-            action(postId);
+        } else {
+            const shouldCallAction =
```

Comment:
Are these changes still needed now that the duplicate `onClick` has been removed from below? I can't 100% remember, but based on my comment [here](https://github.com/mattermost/mattermost/pull/25054#pullrequestreview-1816795377), I think the changes to this method can be removed

Reply:
Yes, it still requires as we also need to handle keydown